### DISCUSSION TITLE
Node repl exit delay

### DISCUSSION THREAD

**Discussion by clibu** [appy-one/acebase#13] [Q&A] [answered]
I've started playing with Acebase and when I run the following code it works as expected but waits for a 1 minute or so timeout before exiting Node. If I don't call ```getStuff()``` or ```putStuff()``` it exits straight away. 

Maybe it is waiting for me to close the db however I can't find any db close function!

````
const { AceBase } = require('acebase');

const db = new AceBase( 'my_db' );     // nodejs

db.ready( () => {
    // Do stuff
    console.log( 'AceBase db.ready()' )

    // putStuff()

    getStuff()
    console.log( 'getStuff() returned' )
});


function putStuff(){
    db.ref('game/config')
        .set({
            name: 'Name of the game',
            max_players: 10
        })
        .then( ref => {
            // stored at /game/config
            console.log( 'ref:', ref )
        })
}


function getStuff(){
    return db.ref( 'game/config' )
        .get( snapshot => {
            if ( snapshot.exists() ){
                let config = snapshot.val();
                console.log( 'config:', config )
            }
            else {
                console.log( 'No snapshot' )
            }
        })
}
````

**Comment by appy-one** [comment]
It's because of the caching of record file address locations which are cleared 1 minute after they are last used. It uses setTimeout to schedule cleanups until cache is empty. After the last setTimeout fired, node's event queue is empty and will exit.

**Reply by appy-one** [reply]
I see I could unref the timeout to fix this, it's worth the try

**Reply by clibu** [reply]
Yes unref or a close db function. In my use case db's will open and close and I'd like to know the close is completed without any delayed timeout ad all resources are free'd.

**Comment by appy-one** [comment, marked as answer]
I added timeout.unref() so it will not prevent the process from exiting when the event loop is empty. I tested thoroughly and found no negative side-effects. Published with v1.2.0

**Reply by clibu** [reply]
Great, I'll try it and report back.

**Reply by clibu** [reply]
Thanks all good now. 

Do I need to close a `ref` that is no longer being used and if so how?

**Reply by appy-one** [reply]
No 😄
